Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2490,12 +2542,16 @@ void CanvasRenderingContext2DBase::putImageData(ImageData& data, int dx, int dy,
     if (!sourceRect.isEmpty())
         buffer->putPixelBuffer(data.pixelBuffer(), sourceRect, IntPoint { destOffset });
 
-    OptionSet<DidDrawOption> options; // ignore transform, shadow, clip, and post-processing
+    if (!layerCount()) {
```

Comment:
We've already returned early at the start of the function if we have open layers, so we don't need this check.